vortex-layout/src/scan/tasks.rs

```diff
@@ -29,6 +29,7 @@ pub type TaskFuture<A> = BoxFuture<'static, VortexResult<A>>;
 pub(super) fn split_exec<A: 'static + Send + Sync>(
     ctx: Arc<TaskContext<A>>,
     split: Range<u64>,
+    limit: Option<&mut usize>,
```

Comment:
there's something weird here - you will still do a bunch of work even if you have `Some(0)` as limit